File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.swift
class Solution {
    func findKthPositive(_ arr: [Int], _ k: Int) -> Int {
        var result: [Int] = []
        var number = 1
        var index = 0
        
        while number < 2000 && index < arr.count {
            if arr[index] != number {
                result.append(number)
            } else {
                index += 1
            }
            
            number += 1
        }
        
        while result.count < k {
            result.append(number)
            number += 1
        }
        
        return result[k-1]
    }
}

File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.py
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        result = []
        index = 0
        current = 1
        
        while index < len(arr) and len(result) < k:
            if current < arr[index]:
                result.append(current)
            else:
                index += 1
            
            current += 1
        
        while len(result) < k:
            result.append(current)
            current += 1
        
        return result[-1]
        